### ISSUE TITLE
Need list of community homes

### ISSUE THREAD

**Issue by benlk** [benlk/harvey-senior-homes#2]
This needs to be as comprehensive as possible a list of senior living facilities, assisted living homes, hospice centers, and senior communities within the path of Hurricane Harvey (all such centers in Texas and Louisiana, essentially. Bonus points for Mississippi) that may be affected by Harvey.

Google Sheet for this data collection is https://docs.google.com/spreadsheets/d/1M0s6mW3SHWslazLE3w4GGLe40ONYbjvLCxibWDhMu2s/

If you'd like to contribute to this issue, please request access to that Google Sheet.

The sheet named `locations` has the following columns:

- `location_name`: Name of location
- `location_owner`: Service provider (Say, Brookdale Senior Living, Genesis HealthCare, HCR Manorcare, **or** an empty string if no service provider is known)
- `id`: unique ID of location: lowercase and dashes, should be recognizably related to the location name and service provider, or the location name if no service provider is known: brookdale-corpus-christie, genesis-woodlands, somerset-retirement-village
- `address`: Street address
- `city`: City/town/county name: what would be used for mailing or driving.
- `county`: County of facility (useful because evac orders are by-county, but harde to automate)
- `state`: state
- `zip`: 5-digit zip code
- `phone`: Contact phone number
- `latitude`: optional, should be latitude of location
- `longitude`: optional, should be longitude of location
- `mpn`: Medicare Provider Number, if known.

The sheet named `locations_statuses` has the following columns:

- `id`: unique ID of location, from the `locations` table
- `date`: day of contact, in YYYY-MM-DD format
- `time`: time of contact, in East-coast time, in 24-hour form HH:MM: 13:32 is 1:32 p.m.
- `color`: status color, determined by whoever contacted the location:
    - green: fully operational
    - yellow: affected by storm but still open: service reductions, canceled activities
    - red: phone lines out, news reports significant damage, flooded, destroyed
    - `evacuated`: certified evacuated
- `status`: status message at contact: this should be something along the lines of "I called at 4:27 p.m. Eastern and received a phone robot message saying that the center was open with full service" or "I called at 5:30 p.m Eastern and the line was dead" or "I called at 7:37 p.m. Eastern and spoke with Mary, who said that they had lost paoer but had a backup generator powering the refrigerators for food and had bottled drinking water available for residents.

**Comment by lisawilliams** [comment]
Ooh. Might be a starting point. Going to go explore. *straps on miner's helmet and switches on lamp* https://data.medicare.gov/data/nursing-home-compare

**Comment by lisawilliams** [comment]
Also: Updated list of cities & towns being evacuated for Harvey: http://www.mysanantonio.com/news/local/article/Here-are-all-the-places-being-evacuated-due-to-11956084.php

**Comment by benlk** [comment]
Issue description has been updated with sheet link, column keys.

**Comment by lisawilliams** [comment]
Excellent. I started work before I read your most recent comment, but you can follow the trail of breadcrumbs from the commit/issues I've linked here. I'm starting out by using Medicare.gov data, and paring that down to the list of counties and cities with evacuation orders. Even after I do that, it will still need to be matched up with the particular data format you've laid out above. I'll be working with .csv files in the /data folder of my own fork of the repo until I get things into better shape, with the idea that the data will eventually live in the Google Sheet you've linked here. 

One suggestion: it turns out facilities have unique ids in the form of Medicare Provider Numbers.  It might be useful to use these because if we do, we could then add an extra layer of information -- nursing homes and assisted living facilities that have been fined for violations of federal guidelines.

**Comment by benlk** [comment]
Oooh, Medicare Provider Numbers is an excellent idea! I'll add a column for that in the sheet, but I think that the core IDs used for matching between the two tables and for the URLs of the per-site status pages should be human-readable.

I ended up splitting `address` into `address`, `city`, `state`, and `zip`.

**Comment by lisawilliams** [comment]
OK. I am continuing to work on the dataset -- I'm using the census geocoder to match county names with our city/state pairs so we can match it up with evacuation orders.  Lots more detail in the commit message above.

**Comment by lisawilliams** [comment]
psst @benlk take a look at the most recent commit message ^^

**Comment by lisawilliams** [comment]
I'm going to take a little break and then return to this. Ben, you can feel free to grab any useful files from my fork of the repo, but let me know if you're working on it. If I don't see a note here, when I return, I'm going to continue working on it. Unless something better occurs to me, I may take a more manual approach -- just getting a list of the names of cities in evacuating counties, sifting and sorting my csv, and copy/pasting facilities into the Google Sheet.

**Comment by benlk** [comment]
Sorry for the late reply - thank you for trying the geocoding stuff. It's not essential; just a nice-to-have in case this ends up including a map of affected locations.

If you put the data in the sheet now, we can start doing county stuff by hand, or see if someone wants to tackle adding counties.

**Comment by lisawilliams** [comment]
@benlk Howdy! Hey, I now have an updated and geocoded list of nursing homes in areas with formal evacuation notices. It turns out to be a relatively small number of facilities (32). The largest chunk are located in Corpus Christi. We would get a lot more if we included Houston, of course, but as far as I can tell, Houston is not under a formal evacuation order.  Let me know if you want to include cities that are in the path but not under evacuation and I can take a look at what might be involved in generating a list of facilities for that.

**Comment by lisawilliams** [comment]
PS: Take a look at my most recent commit message for more details. There are some questions about what to include in "facility owner" column.

**Comment by benlk** [comment]
Yeah, I wanted to include pretty much everything that's going to get rained on this week, to include all of Texas and Louisiana, and maybe parts of Mississippi. At a minimum, we should include the Houston area, as that's part of the most-rain area on the weather maps. But this group is a good start.

> https://github.com/lisawilliams/harvey-senior-homes/commit/89de8d7aef69d4f370404d4d15b212b468ef54ae

Your concerns about "facility owner" are valid; I had thought that most places would have just one owner but that is now proven to be wrong. We could limit that to majority-owner companies, maybe?It would need some caveats on the reader-facing side, for sure.

As an aside, I love how thorough your commit messages are.

**Comment by lisawilliams** [comment]
WAR AND PEACE as commit messages :D 

I can dig in to Houston. That will build the list quite a bit. It would actually help to have someone from Texas sanity-check the list.

**Comment by lisawilliams** [comment]
OK, have started to build out more. Google sheet now contains all of Houston, which triples the size of the list. More detail in the commit message.

**Comment by benlk** [comment]
This list doesn't have Corpus Christi homes that are known to be evacuated (from http://newsroom.brookdale.com/)  - maybe they don't have FPNs?

You had a bunch of numeric IDs in the 'id' column; I copied those to the 'mpn' column and then prefixed those IDs with 'fpn-' because of a problem encountered in creating per-location views (https://github.com/benlk/harvey-senior-homes/issues/10) where the copytext library was returning those numbers as decimals: `455999` became `455999.0`, and it was becoming excessively difficult to get that to return to being `455999`.

**Comment by benlk** [comment]
Note that rows without a 'id' entry won't show up in the app.

**Comment by benlk** [comment]
Change to table schema: status color `grey` is now `evacuated`

**Comment by benlk** [comment]
Adding https://www.brookdale.com/en/find-a-community.html?location=Texas&distance=250&count=125, and none of the 5 seen so far have appeared in the Medicare data at https://data.medicare.gov/Nursing-Home-Compare/Ownership/y2hd-n93e

**Comment by lisawilliams** [comment]
OK, as of now, all entries in the sheet have phone numbers.

**Comment by benlk** [comment]
WaPo has a map in this article with a ton of nursing home locations, but no idea of the data source: https://www.washingtonpost.com/graphics/2017/national/harvey-houston-flooding/
